### PR TITLE
ADD cli test in github actions

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: -cli test run in travis, but always, even if unchanged-

## What I did
I changed the github actions workflow so that there's a workflow for the cli tests

### CODE REVIEW

**Review by Hypnosphi — CHANGES_REQUESTED** [review]
Actually, there's much sense in running those tests on changes in `core` and other packages. There are ways to make the output of cli broken without touching cli code. Basically, those tests can discover hidden breaking changes

File: .github/workflows/tests-cli.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: CLI tests
+
+on: 
+  push:
+    paths:
+    - 'lib/cli/*'
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply .github/workflows/tests-cli.yml]: I tried, don't think they are unfortunately

File: .github/workflows/tests-cli.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - 'lib/cli/*/*/*/*/*/*/*/*/*/*/*'
+
+jobs:
+  build:
+
+    name: Test on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node-version: [10]
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply .github/workflows/tests-cli.yml]: Node is is EOL jan 2020

I could change it to 8 if you'd like

File: lib/cli/README.md [hunk 3]
```diff
@@ -54,3 +54,5 @@ npx -p @storybook/cli sb init --type <type>
 ```
 
 Where type is one of the project types defined in [project_types.js](https://github.com/storybookjs/storybook/blob/master/lib/cli/lib/project_types.js)
+
+<!-- dummy change to test cli action -->
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply lib/cli/README.md]: yes

File: .github/workflows/tests-cli.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .github/workflows/tests-cli.yml]: We can only change node support in major releases

**Comment by ndelangen** [comment]
@Hypnosphi I read you loud and clear. I've changed the config so it only runs on if there's a change in /lib or /app

I don't think a change in /addons would ever require a run of the cli tests.. Do you agree?

@shilman fixed

**Comment by Hypnosphi** [comment]
> I don't think a change in /addons would ever require a run of the cli tests.. Do you agree?

Sure. We don't have any pre-installed addons, right?

**Review by Hypnosphi — APPROVED** [review]

**Comment by ndelangen** [comment]
> Sure. We don't have any pre-installed addons, right?

addon links & actions are installed by default when you use `sb init`

**Comment by ndelangen** [comment]
I had to disable the paths filtering for now because:
https://github.community/t5/GitHub-Actions/Preserve-status-from-previous-action-run/m-p/33821#M1661

**Comment by ndelangen** [comment]
I've removed the cache from travis, it spends ages storing those, I wonder whether it's a net loss.

**Comment by ndelangen** [comment]
Depending on which is faster GitHub actions or Travis, I'll keep either one for these tests.

If GitHub actions is faster, I might repurpose travis to do some other CI stuff.

**Comment by ndelangen** [comment]
Screenshot 2019-10-04 at 14 58 41

VS

Screenshot 2019-10-04 at 14 59 00

**Comment by ndelangen** [comment]
GitHub as a few minutes faster, sadly the tests will have to run on all commits, merging this, and disabling Travis for now. I'll repurpose it for something at some point

**Pull request merged**